plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LibvirtStorageAdaptor.java

```diff
@@ -868,7 +868,7 @@ public KVMPhysicalDisk createPhysicalDisk(String name, KVMStoragePool pool,
 
         StoragePoolType poolType = pool.getType();
         if (poolType.equals(StoragePoolType.RBD)) {
-            return createPhysicalDiskByLibVirt(name, pool, PhysicalDiskFormat.RAW, provisioningType, size);
+            return createPhysicalDiskByQemuImg(name, pool, PhysicalDiskFormat.RAW, provisioningType, size, passphrase);
```

Comment:
Can we check pool details here ?
If pool is an erasure code pool, create volume using qemu-img, otherwise use the current behaviour ?

Reply:
Sure, I added a conditional inside this method to maintain the current behaviour.

Reply:
good, thanks